File: src/main/java/org/dedda/bratwurst/lang/Push.java
package org.dedda.bratwurst.lang;

import org.dedda.bratwurst.lang.scope.Scope;

/**
 * Created by dedda on 12/8/15.
 *
 * @author dedda
 */
public class Push extends BWInstruction {

    private BWExpression argument;

    public Push(int lineNumber, BWExpression argument) {
        super(lineNumber);
        this.argument = argument;
    }

    @Override
    public void run(Scope scope) {
        argument.run(scope);
        scope.push(argument.getValue());
    }

    public BWExpression getArgument() {
        return argument;
    }
}

File: src/main/java/org/dedda/bratwurst/lang/Push.kt
package org.dedda.bratwurst.lang

import org.dedda.bratwurst.lang.scope.Scope

/**
 * Created by dedda on 12/8/15.
 *
 * @author dedda
 */
class Push(lineNumber: Int, val argument: BWExpression) : BWInstruction(lineNumber) {
    override fun run(scope: Scope) {
        argument.run(scope)
        scope.push(argument.value)
    }

}